### ISSUE TITLE
supporting `tidymodels` objects throughout `easystats`

### ISSUE THREAD

**Issue by IndrajeetPatil** [easystats/easystats#165]
It has been a long term goal since the early days of eaystats, but the main blocker is that none of the core contributors have experience with these models... So we're essentially waiting for a willing future someone that would contribute its expertise and motivation by kick-starting the support for more ml models :)

_Originally posted by @DominiqueMakowski in https://github.com/easystats/performance/issues/339#issuecomment-884082743_

**Comment by DominiqueMakowski** [comment]
The first step is essential to support more ML models, then tidymodels is just a wrapper around them so it's not the issue. The issue is supporting models like decision trees, forests, nnets etc. The subsequent problem is that our entire API revolves around concepts like parameters, terms, formulas etc., which doesn't really align perfectly with most of ML models :/

**Comment by pdwaggoner** [comment]
Just to throw my two cents in here, for whatever its worth:

Re: `tidymodels`, that entire approach is a set of wrappers (e.g., `rsample`, `parsnip`, `embed`, and so on). So we would, in effect, be wrapping wrappers. Not sure of the complexity of such a thing... Could be easy, and nice low hanging fruit, especially as more and more R users are migrating to the tidy-approach. But it could be more complex than we are thinking. Further, in this scenario, we'd have to be careful of not reinventing the wheel. E.g., `parameters`, `insight`, etc. are extremely similar to `broom`. So how would the `easystats` approach be distinct? 

Re: ML, I agree about the need for supporting more ML modeling options. You are right that the output, model assessment approaches, etc. are a bit different for ML vs. non-ML approaches to modeling. But in general, there are few (if any) statistical tests as well as no p-values (for the most part). So a lot of what this sort of support would look like would involve model complexity vs. model accuracy, and then extensions for cross-validated frameworks. So for lack of a better way to say it, I think the ML part would include "less stuff" than non-ML, but it would be a very different way of thinking from an architecture stand point.

**Comment by mattansb** [comment]
I agree with @pdwaggoner - I think our efforts would be wasted here, we would wither need to support each underlying model individually (what would `model_parameters` show for non-parametric models??), or we would be extracting information from the `tidymodel` itself, which is already done efficiently by and within the tidymodels ecosystem.

**Comment by mattansb** [comment]
What we can think of doing is adding support in `report` for reporting the CV results / confusion matrix / yardstick results from tidymodels.
